src/main/java/org/broadinstitute/dsm/db/KitType.java

```diff
@@ -88,4 +95,43 @@ public KitType(int kitId, String name, String displayName, boolean manualSentTra
         logger.info("Found " + kitTypes.size() + " kitTypes ");
         return kitTypes;
     }
+
+    public static List<String> getUploadReasons(@NonNull String realm){
+        SimpleResult results = inTransaction((conn) -> {
+            List<String> uploadReasons = new ArrayList<>();
+            SimpleResult dbVals = new SimpleResult();
+            String query = SQL_SELECT_UPLOAD_REASONS;
+
+            try (PreparedStatement stmt = conn.prepareStatement(query)) {
+                stmt.setString(1, realm);
+                try (ResultSet rs = stmt.executeQuery()) {
+                    while (rs.next()) {
```

Comment:
if you use the while here you will overwrite l115 with the next find

Reply:
One pattern we should adopt in various database operations is passing through the `Connection` to these sorts of methods so that we don't run into transactional weirdness.  If each db operation operates in its own connection, updates in one method can commit when the intent is to use a single transaction throughout a single web request.